robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Tuple
+import os
+from robotframework_ls.impl.protocols import IVariableFound
+from robocorp_ls_core.robotframework_log import get_logger
+
+log = get_logger(__name__)
+
+
+class VariablesFromVariablesFileLoader:
+    def __init__(self, path: str):
+        self._path = path
+        self._mtime: Optional[float] = None
+        self._variables: Tuple[IVariableFound, ...] = ()
+
+    def __str__(self):
+        return f"[VariablesFromVariablesFileLoader({self._path})]"
+
+    __repr__ = __str__
+
+    def _iter_vars(self, content):
+        start_multiline = ["(", "[", "{"]
+        end_multiline = [")", "]", "}"]
+        multiline = ""
+        multiple_lines = 0
+        for lineno, line in enumerate(content.splitlines()):
+            line = line.strip()
+
+            if line.startswith("#"):
+                continue
+
+            if "#" in line:
+                line = line.split("#", 1)[0]
```

Comment:
Is this correct? Can't the `#` be escaped? Doesn't Robot Framework itself have some API which could be used to read this file?

i.e.: I think the current parsing can be pretty brittle (I'm ok in merging as is though if you don't want to pursue it further).

Reply:
I have looked around without finding an API which can do the parsing. I noticed that comments in multiline variable, dict etc. screwed things up. As far as i have tested and used it this works for now.

Reply:
I think this is actually a regular python file.

You should be able to use: `robotframework_ls.impl.variable_completions_from_py.collect_variables_from_python_ast` to collect the variables from it.

Reply:
How can I call that function when I have path as variable?

Reply:
@fabioz Should i create a RobotDocument of the file to be able to parse it with collect_variables_from_python_ast?

Reply:
I think I have corrected according to comments now. Could you please retrigger tests?

Reply:
@fabioz I think it might pass the tests now, do you mind triggering it?